### DISCUSSION TITLE
Custom resolvers?

### DISCUSSION THREAD

**Discussion by ghost** [littleK0i/SnowDDL#71] [Q&A] [answered]
Hi.  We're working quite a bit with Dynamic Tables, and we are anticipating needing to do "special handling" of state transitions for them and associated resources. Is there a way we can, using singleDB CLI, make custom handling for DTs? I can see some ways to shoehorn it, and we can of course fork and PR. Thoughts?

**Comment by littleK0i** [comment, marked as answer]
You can define custom resolvers and overload existing ones using exactly the same approach as you did with parsers.

But I would not recommend adding any logic which goes out of scope of object management. SnowDDL was designed for commands executed instantly and in parallel. It is not a good fit for something which may run for many hours, like dynamic table refresh. It will be locked in execution, and you won't get final output until everything is done.

Also, I would not recommend using dynamic tables in general. You may get much better results and much more control with any external orchestration tool running plain SQL.

**Reply by ghost** [reply]
Thank you!

**Comment by ghost** [comment]
One followup to this is: Do I need a custom entrypoint if I want to add custom Blueprints? I would expect so because it seems that Custom Blueprint would depend custom resolvers. Yes?

**Reply by littleK0i** [reply]
You do not need a new entrypoint per se, but you definitely need an "app" dervied from `BaseApp` class:
https://github.com/littleK0i/SnowDDL/blob/master/snowddl/app/base.py#L20

For example, SingleDB is a custom "app" which is derived from `BaseApp` and overloads some of its components:
https://github.com/littleK0i/SnowDDL/blob/master/snowddl/app/singledb.py#L21

You may introduce new resolvers by overloading `resolver_sequence` property.

Once you have an app, you may run it as normal Python script. Entrypoint creates the "app" object and calls `.execute()`.
https://github.com/littleK0i/SnowDDL/blob/master/snowddl/app/singledb.py#L250-L252

You may do exactly the same call from plain Python.

**Reply by ghost** [reply]
Thank you
